### PR TITLE
[Event Hubs] add enqueueTime to Process span links

### PR DESCRIPTION
**Pull request by YijunXieMS**
This PR is for issue #9599, which has a dependency on #10765.

### CODE REVIEW

File: sdk/eventhub/azure-eventhub/azure/eventhub/_utils.py [hunk 1]
```diff
@@ -193,12 +193,17 @@ def trace_link_message(events, parent_span=None):
                     if event.properties:
                         traceparent = event.properties.get(b"Diagnostic-Id", "").decode("ascii")
                         if traceparent:
-                            current_span.link(traceparent)
+                            try:
+                                current_span.link(
+                                    traceparent,
+                                    attributes={"enqueuedTime": event.message.annotations.get(PROP_TIMESTAMP)}
+                                )
+                            except TypeError:  # "attributes" is only available after azure-core 1.5
```
Comment (lmazuel) [inline sdk/eventhub/azure-eventhub/azure/eventhub/_utils.py:201]: Your setup.py should ask azure-core>=1.5.0, no need to be backward compatible

File: sdk/eventhub/azure-eventhub/azure/eventhub/_utils.py [hunk 2]
```diff
@@ -193,12 +193,17 @@ def trace_link_message(events, parent_span=None):
                     if event.properties:
                         traceparent = event.properties.get(b"Diagnostic-Id", "").decode("ascii")
                         if traceparent:
-                            current_span.link(traceparent)
+                            try:
+                                current_span.link(
+                                    traceparent,
+                                    attributes={"enqueuedTime": event.message.annotations.get(PROP_TIMESTAMP)}
```
Comment (lmazuel) [inline sdk/eventhub/azure-eventhub/azure/eventhub/_utils.py:199]: Is there a chance that the annotation in None? And then, do we want to annotate {"enqueuedTime": None"}?

Comment (YijunXieMS) [reply sdk/eventhub/azure-eventhub/azure/eventhub/_utils.py]: Nope. All received events have this system property

**Pull request merged**